packages/docusaurus-plugin-sitemap/src/options.ts

```diff
@@ -24,6 +25,9 @@ const PluginOptionSchema = Joi.object({
     .valid(...Object.values(EnumChangefreq))
     .default(DEFAULT_OPTIONS.changefreq),
   priority: Joi.number().min(0).max(1).default(DEFAULT_OPTIONS.priority),
+  ignorePatterns: Joi.array()
+    .items(Joi.object().instance(RegExp))
```

Comment:
Can we allow strings as well, which will be processed with micromatch? We have a `utils.createMatcher` for that.

Reply:
this runs in Node side so why not expose a callback as well (and thus rename to simply "ignore"?)

a callback is always the most flexible option

Reply:
I don't really like callback-based APIs unless we can't implement a good solution otherwise—Webpack loaders don't use callbacks everywhere only because they can. I do prefer serializable APIs if they provide the right level of abstraction. While `ignorePatterns: ["/tags/**"]` is probably as useful as `ignore: (path) => path.startsWith("/tags/")`, the former is more well-understood by everyone and more approachable by those not well-versed with JS.

Reply:
🤷‍♂️ I don't know, it's also more ambiguous because some edge cases might depend on the underlying glob library etc... A callback always behaves explicitly, other apis are just shortcuts.

Not saying that we shouldn't provide shortcuts, but this use-case seems niche enough that a lower-level but more flexible API might be enough